cli/azd/internal/telemetry/telemetry_context.go

```diff
@@ -0,0 +1,50 @@
+package telemetry
+
+import (
+	"context"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry/baggage"
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry/fields"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func getEnvironmentAttributes(env *environment.Environment) []attribute.KeyValue {
+	return []attribute.KeyValue{
+		fields.ObjectIdKey.String(env.GetPrincipalId()),
+		fields.SubscriptionIdKey.String(env.GetSubscriptionId()),
```

Comment:
If we adopt the environment proposal I am outlining, I can imagine cases where we might not have a subscription ID for some operations.  Will that be problematic longer term?

Reply:
Would be a slight inconvenience but would not make telemetry unusable. It would still be possible to inject into context whenever we are able to get the subscription ID, and I also see likely that we would end up having this again if we decouple from `az`.